test/local_test_ckpt_index_file.py

```diff
@@ -94,10 +91,13 @@ def run_worker(args: List[str | int]) -> None:
     else:
         stage()
 
+    # Take an optimization step
+    optimizer.step()
+    ref = deepcopy(stage.submod.state_dict())
+    save_checkpoint(stage, stage.submod, CKPT_DIR)
```

Comment:
What's the reason to move the three lines here? Does this mean the original test case have an issue?

Reply:
save_checkpoint writes the index as well as saving weights. As they were decoupled before, I was writing the index in the last set of lines and the checkpoints further up. Now, I had to bring save_checkpoint up to do the whole thing.